File: frontend/src/api/Listener.ts
import { AxiosInstance } from "axios";

export class ListenerApi {
  private api: AxiosInstance;

  constructor(api: AxiosInstance) {
    this.api = api;
  }

  async turnOnListener(): Promise<boolean> {
    const response = await this.api.post("/listener/on");
    if (response.status !== 200) {
      return false;
    }
    return response.data?.status || false;
  }
  async turnOffListener(): Promise<boolean> {
    const response = await this.api.post("/listener/off");
    if (response.status !== 200) {
      return false;
    }
    return response.data?.status || false;
  }
}

File: backend/api/listener.py
from fastapi import APIRouter

from services.listener import LISTENER

router = APIRouter(prefix="/listener")


router = APIRouter(prefix="/listener")


@router.post('/on')
async def turn_on_listener():
    try:
        LISTENER.start()
        return {"status": True}
    except Exception as e:
        return {"status": False}


@router.post('/off')
async def turn_off_listener():
    try:
        LISTENER.stop()
        return {"status": True}
    except Exception as e:
        return {"status": False}
